### PR TITLE
ENT-3197 | Update webbj74/toggl-php to Guzzle 6.

### PR DESCRIPTION
**Pull request by samlerner**

### CODE REVIEW

File: src/Toggl/Common/TogglClientAuthPlugin.php [hunk 1]
```diff
@@ -72,10 +71,15 @@ public function onRequestBeforeSend(Event $event)
...
+     *
+     * @return \GuzzleHttp\Psr7\MessageTrait $trait
      */
     public function setAuth(Request $request)
     {
-        $request->setAuth($this->username, $this->password);
+        return $request->withHeader(
+          'Authorization',
+          'Basic ' . base64_encode($this->username . ':' . $this->password)
+        );
```
Comment (webbj74) [inline src/Toggl/Common/TogglClientAuthPlugin.php]: Is this plugin class even necessary in Guzzle 6? I thought 'auth' was passed in the config array to the client class?

Comment (samlerner) [reply src/Toggl/Common/TogglClientAuthPlugin.php]: Very good point. I'll look into removing this class and moving the auth into the `Client` config.

File: .travis.yml [hunk 2]
```diff
@@ -1,9 +1,7 @@
 language: php
 
 php:
-  - "5.5"
-  - "5.4"
-  - "5.3"
+  - "7.1"
```
Comment (webbj74) [inline .travis.yml]: Can we add 7.2 too?

File: composer.json [hunk 3]
```diff
@@ -12,8 +12,9 @@
         }
     ],
     "require": {
-        "php": ">=5.3.0",
-        "guzzle/guzzle": "~3.5"
+        "php": "~7.1",
+        "guzzlehttp/guzzle": "^6.3.3",
+        "symfony/event-dispatcher": "^4.1"
```
Comment (webbj74) [inline composer.json]: Do we actually need to depend on event-dispatcher? 

```
$ grep -rih ^use ./{src,test} | sort -u
use GuzzleHttp\Client;
use GuzzleHttp\HandlerStack;
use GuzzleHttp\Handler\MockHandler;
use GuzzleHttp\Psr7\Response as HttpResponse;
use Toggl\Api\Response as ApiResponse;
use Toggl\Api\TogglApiClientV8;
use Toggl\Common\TogglClient;
use Toggl\Common\TogglClientAuthPlugin;
use Toggl\Reports\TogglReportsApiClientV2;
use Toggl\Test\Api\Response\MeTest;
use Toggl\Test\Api\Response\ProjectTest;
use Toggl\Test\Api\Response\ProjectsTest;
use Toggl\Test\Api\Response\WorkspaceUsersTest;
use Toggl\Test\Api\Response\WorkspacesTest;
```

Comment (samlerner) [reply composer.json]: Apparently not.

```
[slerner:~/Sites/toggl-php] composer why symfony/event-dispatcher
webbj74/toggl-php  dev-ENT-3197  requires  symfony/event-dispatcher (^4.1)
```

**Pull request merged**